[BUG] 最新版代码，nl2sql执行错误，应该是graph更新导致的。

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.util.concurrent.CompletionException: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: DeserializationProblemHandler.handleInstantiationProblem() for type `org.springframework.ai.document.Document` returned value of type `java.util.HashMap`
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2757] (through reference chain: com.alibaba.cloud.ai.graph.OverAllState["data"]->java.util.HashMap["TABLE_DOCUMENTS_FOR_SCHEMA"]->java.util.ArrayList[0])
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073) ~[na:na]
	at com.alibaba.cloud.ai.graph.CompiledGraph$AsyncNodeGenerator.next(CompiledGraph.java:1031) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.async.AsyncGenerator$WithEmbed.next(AsyncGenerator.java:154) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.async.AsyncGenerator$WithEmbed.next(AsyncGenerator.java:171) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.async.AsyncGeneratorOperators.forEachAsync(AsyncGeneratorOperators.java:99) ~[classes/:na]
	at com.alibaba.cloud.ai.controller.Nl2sqlForGraphController.lambda$streamSearch$5(Nl2sqlForGraphController.java:181) ~[classes/:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run$$$capture(CompletableFuture.java:1804) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec$$$capture(ForkJoinTask.java:387) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188) ~[na:na]
Caused by: java.lang.RuntimeException: java.util.concurrent.CompletionException: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: DeserializationProblemHandler.handleInstantiationProblem() for type `org.springframework.ai.document.Document` returned value of type `java.util.HashMap`
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2757] (through reference chain: com.alibaba.cloud.ai.graph.OverAllState["data"]->java.util.HashMap["TABLE_DOCUMENTS_FOR_SCHEMA"]->java.util.ArrayList[0])
	at com.alibaba.cloud.ai.graph.utils.TryFunction.apply(TryFunction.java:33) ~[classes/:na]
	at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:684) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:662) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2200) ~[na:na]
	at com.alibaba.cloud.ai.graph.CompiledGraph$AsyncNodeGenerator.evaluateAction(CompiledGraph.java:834) ~[classes/:na]
	... 14 common frames omitted
Caused by: java.util.concurrent.CompletionException: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: DeserializationProblemHandler.handleInstantiationProblem() for type `org.springframework.ai.document.Document` returned value of type `java.util.HashMap`
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2757] (through reference chain: com.alibaba.cloud.ai.graph.OverAllState["data"]->java.util.HashMap["TABLE_DOCUMENTS_FOR_SCHEMA"]->java.util.ArrayList[0])
	at com.alibaba.cloud.ai.graph.CompiledGraph$AsyncNodeGenerator.lambda$evaluateAction$7(CompiledGraph.java:870) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.utils.TryFunction.apply(TryFunction.java:29) ~[classes/:na]
	... 18 common frames omitted
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: DeserializationProblemHandler.handleInstantiationProblem() for type `org.springframework.ai.document.Document` returned value of type `java.util.HashMap`
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2757] (through reference chain: com.alibaba.cloud.ai.graph.OverAllState["data"]->java.util.HashMap["TABLE_DOCUMENTS_FOR_SCHEMA"]->java.util.ArrayList[0])
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:67) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1888) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.DeserializationContext.handleInstantiationProblem(DeserializationContext.java:1412) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:293) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.ValueInstantiator.createFromObjectWith(ValueInstantiator.java:301) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:214) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:541) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1497) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:348) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:220) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:187) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer._deserializeTypedForId(AsPropertyTypeDeserializer.java:170) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromObject(AsPropertyTypeDeserializer.java:136) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromAny(AsPropertyTypeDeserializer.java:240) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.JsonDeserializer.deserializeWithType(JsonDeserializer.java:159) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:363) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:246) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:30) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.std.StdDelegatingDeserializer.deserialize(StdDelegatingDeserializer.java:170) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.jsontype.impl.AsArrayTypeDeserializer._deserialize(AsArrayTypeDeserializer.java:123) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.jsontype.impl.AsArrayTypeDeserializer.deserializeTypedFromArray(AsArrayTypeDeserializer.java:56) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromAny(AsPropertyTypeDeserializer.java:238) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.JsonDeserializer.deserializeWithType(JsonDeserializer.java:159) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap(MapDeserializer.java:625) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:449) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:32) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer._deserializeTypedForId(AsPropertyTypeDeserializer.java:170) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromObject(AsPropertyTypeDeserializer.java:136) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserializeWithType(MapDeserializer.java:492) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:147) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:310) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4931) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3868) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3836) ~[jackson-databind-2.18.3.jar:2.18.3]
	at com.alibaba.cloud.ai.graph.serializer.plain_text.jackson.JacksonStateSerializer.read(JacksonStateSerializer.java:222) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.serializer.plain_text.jackson.JacksonStateSerializer.read(JacksonStateSerializer.java:39) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.serializer.Serializer.bytesToObject(Serializer.java:54) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.serializer.Serializer.cloneObject(Serializer.java:60) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.serializer.StateSerializer.cloneObject(StateSerializer.java:40) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.CompiledGraph.cloneState(CompiledGraph.java:394) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.CompiledGraph.addCheckpoint(CompiledGraph.java:363) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.CompiledGraph$AsyncNodeGenerator.getNodeOutput(CompiledGraph.java:920) ~[classes/:na]
	at com.alibaba.cloud.ai.graph.CompiledGraph$AsyncNodeGenerator.lambda$evaluateAction$7(CompiledGraph.java:867) ~[classes/:na]
	... 19 common frames omitted

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

```markdown
Spring AI Alibaba version(s):
```

### Debug logs

_No response_

### Anything else?

_No response_

--- a/spring-ai-alibaba-graph-core/src/main/java/com/alibaba/cloud/ai/graph/serializer/plain_text/jackson/JacksonStateSerializer.java
+++ b/spring-ai-alibaba-graph-core/src/main/java/com/alibaba/cloud/ai/graph/serializer/plain_text/jackson/JacksonStateSerializer.java
@@ -205,7 +205,8 @@ public abstract class JacksonStateSerializer extends PlainTextStateSerializer {
 
 			documentModule.addDeserializer(docClass, documentDeserializer);
 
-		} catch (ClassNotFoundException e) {
+		}
+		catch (ClassNotFoundException e) {
 			// Spring AI Document class not available, skip registration
 		}
 
@@ -818,7 +819,7 @@ public abstract class JacksonStateSerializer extends PlainTextStateSerializer {
 
 		@Override
 		public Object deserialize(com.fasterxml.jackson.core.JsonParser p,
-								  com.fasterxml.jackson.databind.DeserializationContext ctxt) throws java.io.IOException {
+				com.fasterxml.jackson.databind.DeserializationContext ctxt) throws java.io.IOException {
 			try {
 				// Parse the JSON node from the input stream
 				com.fasterxml.jackson.databind.JsonNode node = p.getCodec().readTree(p);
@@ -831,24 +832,28 @@ public abstract class JacksonStateSerializer extends PlainTextStateSerializer {
 				// Load Document class using reflection to avoid hard dependency
 				Class<?> documentClass = Class.forName("org.springframework.ai.document.Document");
 
-				// Try constructor with id, content and metadata (most complete constructor)
+				// Try constructor with id, content and metadata (most complete
+				// constructor)
 				try {
 					return documentClass.getConstructor(String.class, String.class, java.util.Map.class)
-							.newInstance(id, content, metadata);
-				} catch (Exception e) {
+						.newInstance(id, content, metadata);
+				}
+				catch (Exception e) {
 					// Fallback: try constructor with content and metadata
 					try {
 						return documentClass.getConstructor(String.class, java.util.Map.class)
-								.newInstance(content, metadata);
-					} catch (Exception ex) {
+							.newInstance(content, metadata);
+					}
+					catch (Exception ex) {
 						// Final fallback: content-only constructor
-						return documentClass.getConstructor(String.class)
-								.newInstance(content != null ? content : "");
+						return documentClass.getConstructor(String.class).newInstance(content != null ? content : "");
 					}
 				}
 
-			} catch (Exception e) {
-				// Throw explicit exception instead of creating empty object to maintain data integrity
+			}
+			catch (Exception e) {
+				// Throw explicit exception instead of creating empty object to maintain
+				// data integrity
 				throw new com.fasterxml.jackson.databind.JsonMappingException(p,
 						"Failed to deserialize Spring AI Document: " + e.getMessage(), e);
 			}
@@ -860,76 +865,77 @@ public abstract class JacksonStateSerializer extends PlainTextStateSerializer {
 		private String extractId(com.fasterxml.jackson.databind.JsonNode node) {
 			// Try different field names for document ID
 			return java.util.Optional.ofNullable(node.get("id"))
+				.map(com.fasterxml.jackson.databind.JsonNode::asText)
+				.orElseGet(() -> java.util.Optional.ofNullable(node.get("docId"))
 					.map(com.fasterxml.jackson.databind.JsonNode::asText)
-					.orElseGet(() -> java.util.Optional.ofNullable(node.get("docId"))
-							.map(com.fasterxml.jackson.databind.JsonNode::asText)
-							.orElseGet(() -> java.util.Optional.ofNullable(node.get("documentId"))
-									.map(com.fasterxml.jackson.databind.JsonNode::asText)
-									.orElse(null))); // Return null if no ID found
+					.orElseGet(() -> java.util.Optional.ofNullable(node.get("documentId"))
+						.map(com.fasterxml.jackson.databind.JsonNode::asText)
+						.orElse(null))); // Return null if no ID found
 		}
 
 		/**
 		 * Extract content from JsonNode - supports multiple field names
 		 */
 		private String extractContent(com.fasterxml.jackson.databind.JsonNode node) {
-			// Try different field names for content, similar to SpringAIMessageDeserializer
-			return java.util.Optional.ofNullable(node.get("content"))
-					.map(contentNode -> {
-						// Handle null values properly - return empty string instead of "null"
-						if (contentNode.isNull()) {
-							return "";
-						}
-						return contentNode.asText();
-					})
-					.orElseGet(() -> java.util.Optional.ofNullable(node.get("text"))
-							.map(textNode -> {
-								if (textNode.isNull()) {
-									return "";
-								}
-								return textNode.asText();
-							})
-							.orElseGet(() -> java.util.Optional.ofNullable(node.get("pageContent"))
-									.map(pageContentNode -> {
-										if (pageContentNode.isNull()) {
-											return "";
-										}
-										return pageContentNode.asText();
-									})
-									.orElseGet(() -> {
-										// If node is plain text, use it directly
-										if (node.isTextual()) {
-											return node.asText();
-										}
-										return ""; // Fallback to empty string for content
-									})));
+			// Try different field names for content, similar to
+			// SpringAIMessageDeserializer
+			return java.util.Optional.ofNullable(node.get("content")).map(contentNode -> {
+				// Handle null values properly - return empty string instead of "null"
+				if (contentNode.isNull()) {
+					return "";
+				}
+				return contentNode.asText();
+			}).orElseGet(() -> java.util.Optional.ofNullable(node.get("text")).map(textNode -> {
+				if (textNode.isNull()) {
+					return "";
+				}
+				return textNode.asText();
+			}).orElseGet(() -> java.util.Optional.ofNullable(node.get("pageContent")).map(pageContentNode -> {
+				if (pageContentNode.isNull()) {
+					return "";
+				}
+				return pageContentNode.asText();
+			}).orElseGet(() -> {
+				// If node is plain text, use it directly
+				if (node.isTextual()) {
+					return node.asText();
+				}
+				return ""; // Fallback to empty string for content
+			})));
 		}
 
 		/**
 		 * Extract metadata from JsonNode - handles nested object structure
 		 * @param node The JSON node containing metadata
 		 */
-		private java.util.Map<String, Object> extractMetadata(com.fasterxml.jackson.databind.JsonNode node
-													 ) {
+		private java.util.Map<String, Object> extractMetadata(com.fasterxml.jackson.databind.JsonNode node) {
 			java.util.Map<String, Object> metadata = new java.util.HashMap<>();
 
 			// Extract metadata object if present
 			com.fasterxml.jackson.databind.JsonNode metadataNode = node.get("metadata");
 			if (metadataNode != null && metadataNode.isObject()) {
-				// Iterate through all metadata fields and convert to appropriate Java types
+				// Iterate through all metadata fields and convert to appropriate Java
+				// types
 				metadataNode.fields().forEachRemaining(entry -> {
 					com.fasterxml.jackson.databind.JsonNode value = entry.getValue();
 					if (value.isTextual()) {
 						metadata.put(entry.getKey(), value.asText());
-					} else if (value.isNumber()) {
+					}
+					else if (value.isNumber()) {
 						metadata.put(entry.getKey(), value.numberValue());
-					} else if (value.isBoolean()) {
+					}
+					else if (value.isBoolean()) {
 						metadata.put(entry.getKey(), value.asBoolean());
-					} else if (value.isNull()) {
+					}
+					else if (value.isNull()) {
 						metadata.put(entry.getKey(), null);
-					} else if (value.isObject() || value.isArray()) {
-						// For complex objects, use recursive conversion to HashMap structure
+					}
+					else if (value.isObject() || value.isArray()) {
+						// For complex objects, use recursive conversion to HashMap
+						// structure
 						metadata.put(entry.getKey(), convertToHashMapStructure(value));
-					} else {
+					}
+					else {
 						metadata.put(entry.getKey(), value.toString());
 					}
 				});
@@ -939,7 +945,8 @@ public abstract class JacksonStateSerializer extends PlainTextStateSerializer {
 		}
 
 		/**
-		 * Converts JsonNode to HashMap/ArrayList/primitive structure (same as FallbackObjectDeserializer)
+		 * Converts JsonNode to HashMap/ArrayList/primitive structure (same as
+		 * FallbackObjectDeserializer)
 		 */
 		private Object convertToHashMapStructure(com.fasterxml.jackson.databind.JsonNode node) {
 			if (node == null || node.isNull()) {
@@ -972,5 +979,7 @@ public abstract class JacksonStateSerializer extends PlainTextStateSerializer {
 				return node.asText();
 			}
 		}
+
 	}
+
 }

--- a/spring-ai-alibaba-graph-core/src/test/java/com/alibaba/cloud/ai/graph/serializer/SpringAIDocumentDeserializerTest.java
+++ b/spring-ai-alibaba-graph-core/src/test/java/com/alibaba/cloud/ai/graph/serializer/SpringAIDocumentDeserializerTest.java
@@ -29,406 +29,358 @@ import static org.junit.jupiter.api.Assertions.*;
  */
 class SpringAIDocumentDeserializerTest {
 
-    private JacksonStateSerializer.SpringAIDocumentDeserializer deserializer;
-    private ObjectMapper objectMapper;
+	private JacksonStateSerializer.SpringAIDocumentDeserializer deserializer;
 
-    @BeforeEach
-    void setUp() {
-        objectMapper = new ObjectMapper();
-        deserializer = new JacksonStateSerializer.SpringAIDocumentDeserializer();
-    }
+	private ObjectMapper objectMapper;
 
-    @Test
-    void testDeserializeDocumentWithAllFields() throws Exception {
-        // Arrange
-        String json = """
-        {
-            "id": "doc-123",
-            "content": "This is a test document content",
-            "metadata": {
-                "source": "test-source",
-                "author": "test-author",
-                "score": 0.95
-            }
-        }
-        """;
+	@BeforeEach
+	void setUp() {
+		objectMapper = new ObjectMapper();
+		deserializer = new JacksonStateSerializer.SpringAIDocumentDeserializer();
+	}
 
-        // Act
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(json),
-            objectMapper.getDeserializationContext()
-        );
+	@Test
+	void testDeserializeDocumentWithAllFields() throws Exception {
+		// Arrange
+		String json = """
+				{
+				    "id": "doc-123",
+				    "content": "This is a test document content",
+				    "metadata": {
+				        "source": "test-source",
+				        "author": "test-author",
+				        "score": 0.95
+				    }
+				}
+				""";
 
-        // Assert
-        assertNotNull(result);
-        assertEquals("org.springframework.ai.document.Document", result.getClass().getName());
-        
-        // Verify content through reflection
-        String content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("This is a test document content", content);
-        
-        // Verify ID through reflection
-        String id = (String) result.getClass().getMethod("getId").invoke(result);
-        assertEquals("doc-123", id);
-        
-        // Verify metadata through reflection
-        Map<?, ?> metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
-        assertNotNull(metadata);
-        assertEquals("test-source", metadata.get("source"));
-        assertEquals("test-author", metadata.get("author"));
-        assertEquals(0.95, metadata.get("score"));
-    }
+		// Act
+		Object result = deserializer.deserialize(objectMapper.createParser(json),
+				objectMapper.getDeserializationContext());
 
-    @Test
-    void testDeserializeDocumentWithAlternativeFieldNames() throws Exception {
-        // Arrange
-        String json = """
-        {
-            "docId": "doc-456",
-            "text": "Alternative content field",
-            "pageContent": "Page content field",
-            "properties": {
-                "category": "test",
-                "priority": 1
-            }
-        }
-        """;
+		// Assert
+		assertNotNull(result);
+		assertEquals("org.springframework.ai.document.Document", result.getClass().getName());
 
-        // Act
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(json),
-            objectMapper.getDeserializationContext()
-        );
+		// Verify content through reflection
+		String content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("This is a test document content", content);
 
-        // Assert
-        assertNotNull(result);
-        assertEquals("org.springframework.ai.document.Document", result.getClass().getName());
-        
-        // Should prefer "text" over "pageContent" for content
-        String content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("Alternative content field", content);
-        
-        // Should use "docId" as ID
-        String id = (String) result.getClass().getMethod("getId").invoke(result);
-        assertEquals("doc-456", id);
-    }
+		// Verify ID through reflection
+		String id = (String) result.getClass().getMethod("getId").invoke(result);
+		assertEquals("doc-123", id);
 
-    @Test
-    void testDeserializeDocumentWithMinimalFields() throws Exception {
-        // Arrange
-        String json = """
-        {
-            "content": "Minimal document"
-        }
-        """;
+		// Verify metadata through reflection
+		Map<?, ?> metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
+		assertNotNull(metadata);
+		assertEquals("test-source", metadata.get("source"));
+		assertEquals("test-author", metadata.get("author"));
+		assertEquals(0.95, metadata.get("score"));
+	}
 
-        // Act
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(json),
-            objectMapper.getDeserializationContext()
-        );
+	@Test
+	void testDeserializeDocumentWithAlternativeFieldNames() throws Exception {
+		// Arrange
+		String json = """
+				{
+				    "docId": "doc-456",
+				    "text": "Alternative content field",
+				    "pageContent": "Page content field",
+				    "properties": {
+				        "category": "test",
+				        "priority": 1
+				    }
+				}
+				""";
 
-        // Assert
-        assertNotNull(result);
-        assertEquals("org.springframework.ai.document.Document", result.getClass().getName());
-        
-        String content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("Minimal document", content);
-        
-        String id = (String) result.getClass().getMethod("getId").invoke(result);
-        assertNotNull(id);
-        
-        // Metadata should be empty
-        Map<?, ?> metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
-        assertTrue(metadata.isEmpty());
-    }
+		// Act
+		Object result = deserializer.deserialize(objectMapper.createParser(json),
+				objectMapper.getDeserializationContext());
 
-    @Test
-    void testDeserializeDocumentWithComplexMetadata() throws Exception {
-        // Arrange
-        String json = """
-        {
-            "id": "complex-doc",
-            "content": "Document with complex metadata",
-            "metadata": {
-                "nested": {
-                    "level1": "value1",
-                    "level2": {
-                        "final": "value2"
-                    }
-                },
-                "array": [1, 2, 3],
-                "boolean": true,
-                "number": 42
-            }
-        }
-        """;
+		// Assert
+		assertNotNull(result);
+		assertEquals("org.springframework.ai.document.Document", result.getClass().getName());
 
-        // Act
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(json),
-            objectMapper.getDeserializationContext()
-        );
+		// Should prefer "text" over "pageContent" for content
+		String content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("Alternative content field", content);
 
-        // Assert
-        assertNotNull(result);
-        
-        Map<?, ?> metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
-        assertNotNull(metadata);
-        
-        // Verify complex nested structure
-        Map<?, ?> nested = (Map<?, ?>) metadata.get("nested");
-        assertEquals("value1", nested.get("level1"));
-        
-        Map<?, ?> level2 = (Map<?, ?>) nested.get("level2");
-        assertEquals("value2", level2.get("final"));
-        
-        // Verify array
-        assertTrue(metadata.get("array") instanceof java.util.List);
-        assertEquals(true, metadata.get("boolean"));
-        assertEquals(42, metadata.get("number"));
-    }
+		// Should use "docId" as ID
+		String id = (String) result.getClass().getMethod("getId").invoke(result);
+		assertEquals("doc-456", id);
+	}
 
-    @Test
-    void testDeserializeDocumentWithEmptyContent() throws Exception {
-        // Arrange
-        String json = """
-        {
-            "text": "",
-            "metadata": {}
-        }
-        """;
+	@Test
+	void testDeserializeDocumentWithMinimalFields() throws Exception {
+		// Arrange
+		String json = """
+				{
+				    "content": "Minimal document"
+				}
+				""";
 
-        // Act
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(json),
-            objectMapper.getDeserializationContext()
-        );
+		// Act
+		Object result = deserializer.deserialize(objectMapper.createParser(json),
+				objectMapper.getDeserializationContext());
 
-        // Assert
-        assertNotNull(result);
-        
-        String content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("", content);
-    }
+		// Assert
+		assertNotNull(result);
+		assertEquals("org.springframework.ai.document.Document", result.getClass().getName());
 
-    @Test
-    void testDeserializeDocumentWithNullContent() throws Exception {
-        // Arrange
-        String json = """
-        {
-            "content": null
-        }
-        """;
+		String content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("Minimal document", content);
 
-        // Act
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(json),
-            objectMapper.getDeserializationContext()
-        );
+		String id = (String) result.getClass().getMethod("getId").invoke(result);
+		assertNotNull(id);
 
-        // Assert
-        assertNotNull(result);
-        
-        String content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("", content); // Should fallback to empty string
-    }
+		// Metadata should be empty
+		Map<?, ?> metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
+		assertTrue(metadata.isEmpty());
+	}
 
-    @Test
-    void testDeserializeDocumentWithMissingContent() throws Exception {
-        // Arrange
-        String json = """
-        {
-            "id": "no-content-doc"
-        }
-        """;
+	@Test
+	void testDeserializeDocumentWithComplexMetadata() throws Exception {
+		// Arrange
+		String json = """
+				{
+				    "id": "complex-doc",
+				    "content": "Document with complex metadata",
+				    "metadata": {
+				        "nested": {
+				            "level1": "value1",
+				            "level2": {
+				                "final": "value2"
+				            }
+				        },
+				        "array": [1, 2, 3],
+				        "boolean": true,
+				        "number": 42
+				    }
+				}
+				""";
 
-        // Act
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(json),
-            objectMapper.getDeserializationContext()
-        );
+		// Act
+		Object result = deserializer.deserialize(objectMapper.createParser(json),
+				objectMapper.getDeserializationContext());
 
-        // Assert
-        assertNotNull(result);
-        
-        String content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("", content); // Should fallback to empty string
-        
-        String id = (String) result.getClass().getMethod("getId").invoke(result);
-        assertEquals("no-content-doc", id);
-    }
+		// Assert
+		assertNotNull(result);
 
+		Map<?, ?> metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
+		assertNotNull(metadata);
 
-    @Test
-    void testDeserializeDocumentWithTextualNode() throws Exception {
-        // Arrange
-        String json = "\"Simple text content\"";
+		// Verify complex nested structure
+		Map<?, ?> nested = (Map<?, ?>) metadata.get("nested");
+		assertEquals("value1", nested.get("level1"));
 
-        // Act
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(json),
-            objectMapper.getDeserializationContext()
-        );
+		Map<?, ?> level2 = (Map<?, ?>) nested.get("level2");
+		assertEquals("value2", level2.get("final"));
 
-        // Assert
-        assertNotNull(result);
-        assertEquals("org.springframework.ai.document.Document", result.getClass().getName());
-        
-        String content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("Simple text content", content);
-    }
+		// Verify array
+		assertTrue(metadata.get("array") instanceof java.util.List);
+		assertEquals(true, metadata.get("boolean"));
+		assertEquals(42, metadata.get("number"));
+	}
 
-    @Test
-    void testDeserializeDocumentWithInvalidJson() throws Exception {
-        // Arrange
-        String json = "{invalid json";
+	@Test
+	void testDeserializeDocumentWithEmptyContent() throws Exception {
+		// Arrange
+		String json = """
+				{
+				    "text": "",
+				    "metadata": {}
+				}
+				""";
 
-        // Act & Assert
-        assertThrows(Exception.class, () -> {
-            deserializer.deserialize(
-                objectMapper.createParser(json),
-                objectMapper.getDeserializationContext()
-            );
-        });
-    }
+		// Act
+		Object result = deserializer.deserialize(objectMapper.createParser(json),
+				objectMapper.getDeserializationContext());
 
-    @Test
-    void testDeserializeDocumentWithVariousIdFieldNames() throws Exception {
-        // Test with "id" field
-        String jsonWithId = "{\"id\":\"test-id\",\"content\":\"test\"}";
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithId),
-            objectMapper.getDeserializationContext()
-        );
-        String id = (String) result.getClass().getMethod("getId").invoke(result);
-        assertEquals("test-id", id);
-        
-        // Test with "docId" field
-        String jsonWithDocId = "{\"docId\":\"test-docid\",\"content\":\"test\"}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithDocId),
-            objectMapper.getDeserializationContext()
-        );
-        id = (String) result.getClass().getMethod("getId").invoke(result);
-        assertEquals("test-docid", id);
-        
-        // Test with "documentId" field
-        String jsonWithDocumentId = "{\"documentId\":\"test-documentid\",\"content\":\"test\"}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithDocumentId),
-            objectMapper.getDeserializationContext()
-        );
-        id = (String) result.getClass().getMethod("getId").invoke(result);
-        assertEquals("test-documentid", id);
-        
-        // Test with no ID field
-        String jsonWithoutId = "{\"content\":\"test\"}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithoutId),
-            objectMapper.getDeserializationContext()
-        );
-        id = (String) result.getClass().getMethod("getId").invoke(result);
-        assertNotNull(id);
-    }
+		// Assert
+		assertNotNull(result);
 
-    @Test
-    void testDeserializeDocumentWithVariousContentFieldNames() throws Exception {
-        // Test with "content" field
-        String jsonWithContent = "{\"content\":\"test-content\"}";
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithContent),
-            objectMapper.getDeserializationContext()
-        );
-        String content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("test-content", content);
-        
-        // Test with "text" field
-        String jsonWithText = "{\"text\":\"test-text\"}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithText),
-            objectMapper.getDeserializationContext()
-        );
-        content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("test-text", content);
-        
-        // Test with "pageContent" field
-        String jsonWithPageContent = "{\"pageContent\":\"test-pagecontent\"}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithPageContent),
-            objectMapper.getDeserializationContext()
-        );
-        content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("test-pagecontent", content);
-        
-        // Test preference order: content > text > pageContent
-        String jsonWithAll = "{\"content\":\"content-value\",\"text\":\"text-value\",\"pageContent\":\"pagecontent-value\"}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithAll),
-            objectMapper.getDeserializationContext()
-        );
-        content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("content-value", content);
-        
-        // Test with textual node
-        String textualJson = "\"simple text\"";
-        result = deserializer.deserialize(
-            objectMapper.createParser(textualJson),
-            objectMapper.getDeserializationContext()
-        );
-        content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("simple text", content);
-        
-        // Test with no content field
-        String jsonWithoutContent = "{\"id\":\"test\"}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithoutContent),
-            objectMapper.getDeserializationContext()
-        );
-        content = (String) result.getClass().getMethod("getText").invoke(result);
-        assertEquals("", content);
-    }
+		String content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("", content);
+	}
 
-    @Test
-    void testDeserializeDocumentWithVariousMetadataTypes() throws Exception {
-        // Test with simple metadata
-        String jsonWithSimpleMetadata = "{\"content\":\"test\",\"metadata\":{\"key1\":\"value1\",\"key2\":42}}";
-        Object result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithSimpleMetadata),
-            objectMapper.getDeserializationContext()
-        );
-        Map<?, ?> metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
-        assertEquals("value1", metadata.get("key1"));
-        assertEquals(42, metadata.get("key2"));
-        
-        // Test with nested metadata
-        String jsonWithNestedMetadata = "{\"content\":\"test\",\"metadata\":{\"nested\":{\"inner\":\"value\"}}}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithNestedMetadata),
-            objectMapper.getDeserializationContext()
-        );
-        metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
-        Map<?, ?> nested = (Map<?, ?>) metadata.get("nested");
-        assertEquals("value", nested.get("inner"));
-        
-        // Test with array metadata
-        String jsonWithArrayMetadata = "{\"content\":\"test\",\"metadata\":{\"array\":[1,2,3]}}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithArrayMetadata),
-            objectMapper.getDeserializationContext()
-        );
-        metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
-        assertTrue(metadata.get("array") instanceof java.util.List);
-        
-        // Test with no metadata
-        String jsonWithoutMetadata = "{\"content\":\"test\"}";
-        result = deserializer.deserialize(
-            objectMapper.createParser(jsonWithoutMetadata),
-            objectMapper.getDeserializationContext()
-        );
-        metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
-        assertTrue(metadata.isEmpty());
-    }
+	@Test
+	void testDeserializeDocumentWithNullContent() throws Exception {
+		// Arrange
+		String json = """
+				{
+				    "content": null
+				}
+				""";
 
+		// Act
+		Object result = deserializer.deserialize(objectMapper.createParser(json),
+				objectMapper.getDeserializationContext());
+
+		// Assert
+		assertNotNull(result);
+
+		String content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("", content); // Should fallback to empty string
+	}
+
+	@Test
+	void testDeserializeDocumentWithMissingContent() throws Exception {
+		// Arrange
+		String json = """
+				{
+				    "id": "no-content-doc"
+				}
+				""";
+
+		// Act
+		Object result = deserializer.deserialize(objectMapper.createParser(json),
+				objectMapper.getDeserializationContext());
+
+		// Assert
+		assertNotNull(result);
+
+		String content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("", content); // Should fallback to empty string
+
+		String id = (String) result.getClass().getMethod("getId").invoke(result);
+		assertEquals("no-content-doc", id);
+	}
+
+	@Test
+	void testDeserializeDocumentWithTextualNode() throws Exception {
+		// Arrange
+		String json = "\"Simple text content\"";
+
+		// Act
+		Object result = deserializer.deserialize(objectMapper.createParser(json),
+				objectMapper.getDeserializationContext());
+
+		// Assert
+		assertNotNull(result);
+		assertEquals("org.springframework.ai.document.Document", result.getClass().getName());
+
+		String content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("Simple text content", content);
+	}
+
+	@Test
+	void testDeserializeDocumentWithInvalidJson() throws Exception {
+		// Arrange
+		String json = "{invalid json";
+
+		// Act & Assert
+		assertThrows(Exception.class, () -> {
+			deserializer.deserialize(objectMapper.createParser(json), objectMapper.getDeserializationContext());
+		});
+	}
+
+	@Test
+	void testDeserializeDocumentWithVariousIdFieldNames() throws Exception {
+		// Test with "id" field
+		String jsonWithId = "{\"id\":\"test-id\",\"content\":\"test\"}";
+		Object result = deserializer.deserialize(objectMapper.createParser(jsonWithId),
+				objectMapper.getDeserializationContext());
+		String id = (String) result.getClass().getMethod("getId").invoke(result);
+		assertEquals("test-id", id);
+
+		// Test with "docId" field
+		String jsonWithDocId = "{\"docId\":\"test-docid\",\"content\":\"test\"}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithDocId),
+				objectMapper.getDeserializationContext());
+		id = (String) result.getClass().getMethod("getId").invoke(result);
+		assertEquals("test-docid", id);
+
+		// Test with "documentId" field
+		String jsonWithDocumentId = "{\"documentId\":\"test-documentid\",\"content\":\"test\"}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithDocumentId),
+				objectMapper.getDeserializationContext());
+		id = (String) result.getClass().getMethod("getId").invoke(result);
+		assertEquals("test-documentid", id);
+
+		// Test with no ID field
+		String jsonWithoutId = "{\"content\":\"test\"}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithoutId),
+				objectMapper.getDeserializationContext());
+		id = (String) result.getClass().getMethod("getId").invoke(result);
+		assertNotNull(id);
+	}
+
+	@Test
+	void testDeserializeDocumentWithVariousContentFieldNames() throws Exception {
+		// Test with "content" field
+		String jsonWithContent = "{\"content\":\"test-content\"}";
+		Object result = deserializer.deserialize(objectMapper.createParser(jsonWithContent),
+				objectMapper.getDeserializationContext());
+		String content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("test-content", content);
+
+		// Test with "text" field
+		String jsonWithText = "{\"text\":\"test-text\"}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithText),
+				objectMapper.getDeserializationContext());
+		content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("test-text", content);
+
+		// Test with "pageContent" field
+		String jsonWithPageContent = "{\"pageContent\":\"test-pagecontent\"}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithPageContent),
+				objectMapper.getDeserializationContext());
+		content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("test-pagecontent", content);
+
+		// Test preference order: content > text > pageContent
+		String jsonWithAll = "{\"content\":\"content-value\",\"text\":\"text-value\",\"pageContent\":\"pagecontent-value\"}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithAll),
+				objectMapper.getDeserializationContext());
+		content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("content-value", content);
+
+		// Test with textual node
+		String textualJson = "\"simple text\"";
+		result = deserializer.deserialize(objectMapper.createParser(textualJson),
+				objectMapper.getDeserializationContext());
+		content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("simple text", content);
+
+		// Test with no content field
+		String jsonWithoutContent = "{\"id\":\"test\"}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithoutContent),
+				objectMapper.getDeserializationContext());
+		content = (String) result.getClass().getMethod("getText").invoke(result);
+		assertEquals("", content);
+	}
+
+	@Test
+	void testDeserializeDocumentWithVariousMetadataTypes() throws Exception {
+		// Test with simple metadata
+		String jsonWithSimpleMetadata = "{\"content\":\"test\",\"metadata\":{\"key1\":\"value1\",\"key2\":42}}";
+		Object result = deserializer.deserialize(objectMapper.createParser(jsonWithSimpleMetadata),
+				objectMapper.getDeserializationContext());
+		Map<?, ?> metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
+		assertEquals("value1", metadata.get("key1"));
+		assertEquals(42, metadata.get("key2"));
+
+		// Test with nested metadata
+		String jsonWithNestedMetadata = "{\"content\":\"test\",\"metadata\":{\"nested\":{\"inner\":\"value\"}}}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithNestedMetadata),
+				objectMapper.getDeserializationContext());
+		metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
+		Map<?, ?> nested = (Map<?, ?>) metadata.get("nested");
+		assertEquals("value", nested.get("inner"));
+
+		// Test with array metadata
+		String jsonWithArrayMetadata = "{\"content\":\"test\",\"metadata\":{\"array\":[1,2,3]}}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithArrayMetadata),
+				objectMapper.getDeserializationContext());
+		metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
+		assertTrue(metadata.get("array") instanceof java.util.List);
+
+		// Test with no metadata
+		String jsonWithoutMetadata = "{\"content\":\"test\"}";
+		result = deserializer.deserialize(objectMapper.createParser(jsonWithoutMetadata),
+				objectMapper.getDeserializationContext());
+		metadata = (Map<?, ?>) result.getClass().getMethod("getMetadata").invoke(result);
+		assertTrue(metadata.isEmpty());
+	}
 
 }

